App features finished

Usagib opened the pull request:
- app clasees implemented
- functionality modules implemented
- frontend implemented and imported
- functionality added to frontend elements
- app styling finished
- esLint checks passed
- Webpack(ed) page

Usagib committed:
project and todo classes ADDED

updated controller

Commit:
refactor project and todo classes

Usagib committed:
function structure created

updated index.js

Commit:
create addProject

Commit:
add addTodo function

Commit:
add populateProjects

Commit:
Update index.js

Commit:
create localSave

Commit:
edit removeProject

Commit:
edit removeProject

Commit:
refactor removeProject from id to item # in array

Commit:
add comments to functions

Usagib committed:
addTodo functionality added, rendered frontend elements 

addTodo updated on index.js appended edit and delete buttons

Commit:
add badge to todos

Usagib committed:
added edit funcionality to projects

added modal on home.js added edit event on index.js

Usagib committed:
Merge remote-tracking branch 'origin/feature-05-local-storage' into feature-05-local-storage

Commit:
update todo modal form layout

Commit:
update todo modal select

Commit:
update todo model text area

Usagib committed:
Merge remote-tracking branch 'origin/feature-05-local-storage' into feature-05-local-storage

Commit:
edit home.js to ajust padding on main tables

Usagib committed:
remove todo feature added

updtaed main.js

Commit:
edit badge vertical alignment

Usagib committed:
Merge remote-tracking branch 'origin/feature-05-local-storage' into feature-05-local-storage

Usagib committed:
Merge remote-tracking branch 'origin/feature-05-local-storage' into feature-05-local-storage

Commit:
center badge on table

Commit:
add app screenshot

Usagib committed:
Auto stash before merge of "feature-05-local-storage" and "origin/feature-05-local-storage"

Commit:
change name order on footer

Commit:
Merge branch 'feature-05-local-storage' of https://github.com/Usagib/js-todolist into feature-05-local-storage

Usagib committed:
functions updated

Usagib committed:
Merge remote-tracking branch 'origin/feature-05-local-storage' into feature-05-local-storage

Commit:
debug edit projects

Commit:
Merge branch 'feature-05-local-storage' of https://github.com/Usagib/js-todolist into feature-05-local-storage

Commit:
edit button id editProject

Commit:
refactor edit project function

Commit:
refactor remove todo

Commit:
refactor edit todo function

Commit:
complete refactoring editTodo function

Commit:
edit editProject to render project list

Usagib committed:
setEditModal function added

prefill modals for editing functionality added

Usagib committed:
Merge remote-tracking branch 'origin/feature-05-local-storage' into feature-05-local-storage

Usagib committed:
added screenshot updated readme

Usagib committed:
lintfix-1

updated localsave.js index.js

Usagib committed:
lint-fix2

index.js updated

Usagib committed:
eslint organitation fix

Usagib committed:
final lint fix

Usagib committed:
updated code organization and built solution

added eslint exception

Usagib commented:
Hello Team!
As requested the project features implemented are as follows:
- implemented local storage save
- friendly and navigable UI
- OOP Paradigm applied
- MVC modularization applied
- dinamically generated elements

Commit:
Update main.js

mcrd25 commented on src/index.js:
```diff
@@ -5,107 +5,209 @@ import './view/css/style.css';
 activeInfo.innerHTML = home;
 
+// save to local storage
+function saveLocal() {
```
You ought to use ES6 Syntax for this project:
Which means usage of arrow functions instead of the function keyword.
See this [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/todo_list/es6_requirement.png) from pathwright

mcrd25 commented on src/index.js:
```diff
@@ -5,107 +5,209 @@ import './view/css/style.css';
-  const name = pName;
-  let taskArr = [];
+// add project to projectList, save local and reload
+function addProject() {
```
You ought to use ES6 Syntax for this project:
Which means usage of arrow functions instead of the function keyword.
See this [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/todo_list/es6_requirement.png) from pathwright

mcrd25 commented on src/index.js:
```diff
@@ -5,107 +5,209 @@ import './view/css/style.css';
 
-  const addTask = (Arr, task) => {
-  }
+function editProject(projectIndex) {
```
You ought to use ES6 Syntax for this project:
Which means usage of arrow functions instead of the function keyword.
See this [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/todo_list/es6_requirement.png) from pathwright

mcrd25 commented on src/index.js:
```diff
@@ -5,107 +5,209 @@ import './view/css/style.css';
-    this.name = newName;
-  }
+// remove project from projectList, save local and render
+function removeProject(project) {
```
You ought to use ES6 Syntax for this project:
Which means usage of arrow functions instead of the function keyword.
See this [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/todo_list/es6_requirement.png) from pathwright

mcrd25 commented on src/index.js:
```diff
@@ -5,107 +5,209 @@ import './view/css/style.css';
+}
 
-  }
+function setProjectModal(projectIndex) {
```
You ought to use ES6 Syntax for this project:
Which means usage of arrow functions instead of the function keyword.
See this [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/todo_list/es6_requirement.png) from pathwright

mcrd25 commented on src/index.js:
```diff
@@ -5,107 +5,209 @@ import './view/css/style.css';
-    getName,
-    taskArr
-  };
+function removeTodo(project, todo) {
```
You ought to use ES6 Syntax for this project:
Which means usage of arrow functions instead of the function keyword.
See this [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/todo_list/es6_requirement.png) from pathwright

mcrd25 commented on src/index.js:
```diff
@@ -5,107 +5,209 @@ import './view/css/style.css';
-    getDate,
-    getChecked
-  };
+function editTodo(todoIndex) {
```
You ought to use ES6 Syntax for this project:
Which means usage of arrow functions instead of the function keyword.
See this [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/todo_list/es6_requirement.png) from pathwright

mcrd25 commented on src/index.js:
```diff
@@ -5,107 +5,209 @@ import './view/css/style.css';
 }
 
-const render = () => {
+function setTodoModal(todoIndex) {
```
You ought to use ES6 Syntax for this project:
Which means usage of arrow functions instead of the function keyword.
See this [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/todo_list/es6_requirement.png) from pathwright

mcrd25 commented on src/index.js:
```diff
@@ -5,107 +5,209 @@ import './view/css/style.css';
+}
 
+// add todo to project save local and render
+function addTodo(todoList) {
```
You ought to use ES6 Syntax for this project:
Which means usage of arrow functions instead of the function keyword.
See this [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/todo_list/es6_requirement.png) from pathwright

mcrd25 commented on src/index.js:
```diff
@@ -5,107 +5,209 @@ import './view/css/style.css';
+btnEditTodo.onclick = () => { editTodo(btnEditTodo.getAttribute('curr-todo-index')); };
+
+// render todos table
+function renderTodos(project) {
```
You ought to use ES6 Syntax for this project:
Which means usage of arrow functions instead of the function keyword.
See this [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/todo_list/es6_requirement.png) from pathwright

mcrd25 commented on src/index.js:
```diff
@@ -5,107 +5,209 @@ import './view/css/style.css';
-const test = () => {
-  let newProject = project('restaurant page');
+// render projectsList to project table
+function renderProjects(projectList) {
```
You ought to use ES6 Syntax for this project:
Which means usage of arrow functions instead of the function keyword.
See this [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/todo_list/es6_requirement.png) from pathwright

mcrd25 commented on src/model/localSave.js:
```diff
@@ -0,0 +1,30 @@
+
+let localProjectList = []; // eslint-disable-line import/no-mutable-exports
+
+function populateProjects() {
```
You ought to use ES6 Syntax for this project:
Which means usage of arrow functions instead of the function keyword.
See this [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/todo_list/es6_requirement.png) from pathwright

mcrd25 reviewed (changes requested):
# Changes Required 🔍 

Hi Team! 👋 

Good job! I really like y'all git-flow and the use of multiple feature branches. 😄 


##### Required changes
- You should have both a meaningful title and description of your PR (should reflect what was accomplished) See [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/descriptive_prs.png) from Pathwright
- Please use ES6 syntax [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/todo_list/es6_requirement.png)

##### Optional changes [OPTIONAL]
- I recommend adding a `start` script to package.json your Webpack Dev Server
- User should not be able to enter empty project name or empty TODO details 
- Consider figuring out a way to keep todo list loaded when person adds/removes or changes an item in the project
- Consider having clear way to assign todo to project, as it stands I am able to not select a project and create a TODO which does not show up anywhere
***Please makes changes and submit another review request.
If you have any questions, let me know here or on slack and I will be glad to clarify with you. You can also reach me and connect with me on my social media or email.***

*Happy coding 😄,*
**[--Maya](https://github.com/mcrd25)**
[GitHub](https://github.com/mcrd25) [Twitter](https://twitter.com/mcrd25) [linkedin](https://www.linkedin.com/in/mayadouglas/) [GitLab](https://gitlab.com/mcrd25)

---
*P.S. all requested changes not marked by [OPTIONAL] or are not positive comments are **REQUIRED***

Commit:
Merge branch 'feature-05-local-storage' of https://github.com/Usagib/js-todolist into feature-05-local-storage

Commit:
refactor index.js to use arrow functions instead of the function keyword

Commit:
refactor localSave.js to arrow function

Usagib committed:
added webpack dev server to start script

package.json updated

Usagib committed:
esLint fix

Usagib committed:
Validation to form added, updated ui

updated index.js

Usagib committed:
esLint Fix

Usagib committed:
minor esLint fixes

Usagib commented:
Hello Team!
We've made the proper changes to the app: they are as follow:
- changed syntax to use ES6 arrow functions
- added form validations
- changed new todo button to be more accesible to user
- added start script to package.json

Additional comments:
- The app changes are always loaded due to the localSave function implemented

Thank you for the insight provided
Sending PR for code review

maelfosso reviewed (changes requested):
# Changes required :red_circle: 

Hello, Thank you for the improvements you've done.
However, there are still a bug to fix.

Clicking on `Edit` for a TODO obviously open a modal but doesn't load the DUE DATE. Please, fix that. 
Notice that, the bug only happen with the Todo populated not with the one I created when testing the project.

Kindly fix that bugs and submit a new code review request
Thanks

Mael FOSSO

Usagib committed:
prepopulated date fix

Usagib commented:
Hello Team!, thank you for the feedback, the following changes had been implemented:
- Fixed error where prepopulated task date wasn't present on edit modal

Redvanisation reviewed (approved):
# Project approved ✔️ 

### Hello team 🌟 

### _Nice work on this project, it seems that you have met the requirements. 👍_ 

### You may merge to master and submit a project completion form. 🚀 

### Good luck! 🍀
